Add a website for documentation on WFSim

jorana opened the pull request:
# Add a website for WFSim documentation.

In this PR:

- I set up a website for WFSim using similar infrastructure to straxen.
- Remove some notebooks that are so out of date that it makes no sense to keep them around.
- Set up a test website to look at the documentation:
  - https://wfsim-test.readthedocs.io/en/latest

## How does this work
Using https://www.sphinx-doc.org and https://readthedocs.org/ a website can be set up in a similar fashion to strax and straxen. I at first did it without the straxen-way but I think it makes sense to abide to a common format so I copied quite some functionality.

## Auto-generated documentation 
[WFSim documentation.pdf](https://github.com/XENONnT/WFSim/files/6054273/document.pdf)


## Further work
- Add a badge to the website
- Remove the other notebooks not linked on the website or add those notebooks
- Mention NV integration
- Change https://wfsim-test.readthedocs.io/en/latest to https://wfsimreadthedocs.io/en/latest

mzks commented:
Thank you very much @jorana ! It's very useful for users, that's what I wanted actually.
I would like to add few minor comments.

 - Install
 I guess some developers are interested in the documents. It's better to add the developing guides. As a one of examples, like that in install part
 ```
 git clone ...
 pip install -e ./wfsim --user
 ```
 and something.
 
  - working examples
  The detail documents which explain how wfsim works is very nice. If possible, I would like to see the working example at the initial part of the tutorial. I think it is also useful in particular for beginners like me. In addition, to show how to switch some parameters is much helpful.

 - Gain & Gain Spread
  I guess people who will read the document want to change some parameters. I think it is great to show these parameter format and value units. (I just want to know them...)

 - NV (Optical) Usage
  I think it is better to make another pull-request for that after merging. Thank you very much again for your detail explanations!

WenzDaniel commented:
Thanks a lot Joran. Looks like the usual docs magic. I was not aware that one can also use ipynb within the sphinx syntax, nice.

jorana commented:
@mzks , thanks! I agree with all your points, let me add it after getting some more input from especially the core dev. team.

petergaemers commented on docs/source/index.rst:
```diff
@@ -0,0 +1,53 @@
+
+    tutorials/Understanding_WFSim_P1.ipynb
+    tutorials/Understanding_WFSim_P2.ipynb
+    tutorials/Advanced_tricks.ipynb
```
Can you remove this guy and replace it with Fax_tutorial.ipynb

This advanced_tricks guy is secretly very outdated

ramirezdiego commented:
This is great, thanks @jorana. Can we take the change to update the `Fax_Tutorial.ipynb` (and change its name, btw), with all the options one can feed to the context, the default for the nTsims one, etc.?

jorana commented:
Alright, I've done most of the comments in this thread. I would like explicit approval by @zhut19 and @petergaemers because I am deleting there old notebooks:

- [ ] Basics.ipynb
- [ ] bias_smearing demo
- [ ] test_simulator

I also renamed the Fax-tutorial to getting started with wfsim, fixed some spelling and headers there.

ramirezdiego commented on README.md:
```diff
@@ -1,5 +1,7 @@
-# Fax in strax
+# WFSim
+## Fax in strax
```
Thanks! My OCD is better today.
